lib/Http/HtmlResponse.php

```diff
@@ -25,29 +25,40 @@
 
 namespace OCA\Mail\Http;
 
+use OCP\Util;
 use OCP\AppFramework\Http\Response;
 
 class HtmlResponse extends Response {
 
 	/** @var string */
 	private $content;
 
-	private $injectedStyles = <<<EOF
-* { font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Cantarell, Ubuntu, 'Helvetica Neue', Arial, 'Noto Color Emoji', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol'; }
-EOF;
+	/** @var bool */
+	private $plain;
 
-
-	public function __construct(string $content) {
+	/**
+	 * @param string $content message html content
+	 * @param bool $plain do not inject scripts if true (default=false)
+	 */
+	public function __construct(string $content, bool $plain=false) {
 		parent::__construct();
 		$this->content = $content;
+		$this->plain = $plain;
 	}
 
 	/**
-	 * Simply sets the headers and returns the file contents
+	 * Inject scripts if not plain and return message html content.
 	 *
-	 * @return string the file contents
+	 * @return string message html content
 	 */
 	public function render(): string {
-		return '<style>' . $this->injectedStyles . '</style>' . $this->content;
+		if ($this->plain) {
+			return $this->content;
+		}
+
+		$nonce = \OC::$server->getContentSecurityPolicyNonceManager()->getNonce();
```

Comment:
just noticed the static methods. guess we should require the nonce manager and the url generator as constructor arg

Reply:
but I'm fine doing this in a follow-up

Reply:
Sure. Just noticed that `getContentSecurityPolicyNonceManager()` is deprecated. I still have so much to learn about php :D

Reply:
that is because we are in the process of cleaning up some old code https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/upgrade-guide.html#deprecated-apis :)